guides/howtos/test-using-mox.md

```diff
@@ -0,0 +1,88 @@
+# Test Using Mox with Tesla
+
+To mock HTTP requests in your tests using Mox with the Tesla HTTP client,
+follow these steps:
+
+## 1. Define a Mock Adapter
+
+First, define a mock adapter that implements the Tesla.Adapter behaviour. This
+adapter will intercept HTTP requests during testing.
+
+Create a file at `test/support/mocks.ex`:
+
+```elixir
+# test/support/mocks.ex
+Mox.defmock(MyApp.MockAdapter, for: Tesla.Adapter)
+```
+
+## 2. Configure the Mock Adapter for Tests
+
+In your `config/test.exs` file, configure Tesla to use the mock adapter you
+just defined:
+
+```elixir
+# config/test.exs
+config :tesla, adapter: MyApp.MockAdapter
+```
+
+If you are not using the global adapter configuration, ensure that your Tesla
+client modules are configured to use `MyApp.MockAdapter` during tests.
+
+## 3. Set Up Mocking in Your Tests
+
+Create a test module, for example `test/demo_test.exs`, and set up `Mox` to
+define expectations and verify them:
+
+```elixir
+defmodule MyApp.FeatureTest do
+  use ExUnit.Case, async: true
+
+  require Tesla.Test
+
+  setup context, do: Mox.set_mox_from_context(context)
+  setup context, do: Mox.verify_on_exit!(context)
+
+  test "example test" do
+    # Expect a single HTTP request to be made and return a JSON response
+    Tesla.Test.expect_tesla_call(
```

Comment:
Is it necessary to repeat "tesla" in the function name? 
I think `Tesla.Test.expect_call` or just `Tesla.Test.expect` would do.

Reply:
Usually. I put importance on the functions over the module.

If people prefer `import Tesla.Test` then they will avoid collisions with 
Other assertions are available if people prefer to import the assertions functions. Also, "call" makes it clear what that one is about.

Reply:
I'd say we should either:
- Option A: Keep `expect_tesla_call` and `import Tesla.Test` in all examples
- Option B: Change to `Tesla.Test.expect` and never use imports

Reply:
Doing Option A; is what I had in mind, it is a matter of time 😄

Reply:
done